scripts/settings.gradle.kts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2021 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+pluginManagement {
+    resolutionStrategy {
+        eachPlugin {
+            if (requested.id.id == "com.netflix.dgs.codegen") {
```

Comment:
What does CODEGEN_SNAPSHOT resolve to?

Reply:
This file is used as a template, the actual version is _inferred_ via Gradle's Project task and done via the `infer_version` function in the script. Then the `<<<CODEGEN_SNAPSHOT>>>` is replaced with the proper version as part of the function that sets the `settings.gradle.kts` file for the project.

```
def infer_version():
    regex = re.compile(r"Inferred project: graphql-dgs-codegen, version: ([0-9A-Z\-.]+)")
    out = subprocess.check_output([gradlew, "-p", gradle_wd, "project"]).decode("utf-8")
    Out.debug(f"Process output:\n{out}", State.verbose)
    match = re.search(regex, out)
    return match.group(1) if match else ""
```